database-mongo/index.js

```diff
@@ -13,22 +13,117 @@ db.once('open', function() {
 });
 
 var games = mongoose.Schema({
-  token: String,
-  players: String,
-  results: String  
+  token: {type: String, required: true},	
+  playerRoles: String,
+  results: String, 
+  playerIds: String,
 });
 
-var Game = mongoose.model('Game', games);
+Game = mongoose.model('Game', games);
 
-var selectGame = function(token, callback) {
-  Game.find({token});
+module.exports.selectGame = function(token, callback) {
+  Game.find({token}, (err, game) => {
+  	callback(game[0]); //game object with token, results, playerIdMapping, playerRolesMapping etc.
+  });
 };
 
-var createGame = function(token, players, results, callback) {
+module.exports.createGame = function(token, hostName, clientId, callback) {
+	Game.create({token}, (err) => {
+		if (err) {
+			callback(err);
+		} else {
+			var playerIds = {};
+			playerIds[hostName] = clientId;
+			playerIds = JSON.stringify(playerIds);
+			var results = [];
+			results = JSON.stringify(results);
+			Game.update({token}, {playerIds, results}, (err) => {
+				callback(err);
+			});
+		}
+	});
 };
 
-var updateResults = function (results, callback) {
+module.exports.addPlayer = function(token, playerName, clientId, callback) {
+	Game.find({token}, (err, game) => {
+		var playerIds = JSON.parse(game[0].playerIds);
```

Comment:
Is the `game[0]` here and at line 26 to guard against multiple games being returned on a Game.find? If so, is that the right thing? The odds of a token collision are very low. But, were there one, wouldn't this result in two (or more) games worth of players all attempting to play in the same game? Or, is it that mongoose always returns an array of results on a search?